pkg/cgroups/manager/manager.go

```diff
@@ -142,6 +160,7 @@ func (m *CgroupManager) DumpPodMetadata() []*FullPodMetadata {
 	}
 	allMetaOut := make(chan []*FullPodMetadata)
 
+	m.wg.Add(1)
```

Comment:
`DumpPodMetadata()` and `GetPodMetadataForContainer()` already wait for the event to complete before returning the result. No need for changes to these functions.